ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/AggregateAnalyzer.java

```diff
@@ -300,6 +311,19 @@ private AggregateVisitor(
       this.dereferenceCollector = requireNonNull(dereferenceCollector, "dereferenceCollector");
     }
 
+    @Override
+    public Void process(final Expression node, final Void context) {
+      if (groupBy.contains(node) && currentlyInExpressionPartOfGroupBy == null) {
+        currentlyInExpressionPartOfGroupBy = new Pair<>(node, true);
```

Comment:
do we ever actually use the boolean? it looks like we proxy for it by checking `!= null`

Reply:
You are right, will change it. The initial intention was to use it